Week4/week 4.ipynb

```diff
@@ -0,0 +1,335 @@
+   "source": [
+    "def create_contact_table():\n",
+    "    tbl=\"Create table IF NOT EXISTS contact(\\\n",
+    "        fname text,\\\n",
+    "        lname text,\\\n",
+    "        contact integer,\\\n",
+    "        email text,\\\n",
+    "        city text\\\n",
+    "        )\"\n",
+    "    curobj.execute(tbl)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "f597a134",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def create_log_table():\n",
+    "    logtbl=\"create table  IF NOT EXISTS logtable1\\\n",
+    "        (\\\n",
+    "        fname text,\\\n",
+    "        lname text,\\\n",
+    "        contact integer,\\\n",
+    "        datetime text,\\\n",
+    "        operation text\\\n",
+    "        );\"\n",
+    "    curobj.execute(logtbl)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "079f160a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    " def contact_validate_trigger():\n",
+    "    con_validate_trigger=\"create trigger IF NOT EXISTS contact_validate\\\n",
+    "                        before insert\\\n",
+    "                        on contact\\\n",
+    "                        begin\\\n",
+    "                        select\\\n",
+    "                                case \\\n",
+    "                                when length(new.contact)>10 then\\\n",
+    "                                raise(abort,'Given number is greater than 10 digit . Please enter 10 digit number')\\\n",
+    "                                when length(new.contact)<10 then\\\n",
+    "                                raise(abort,'Given number is less than 10 digit . Please enter 10 digit number')\\\n",
+    "                        end;\\\n",
+    "                        end;\"\n",
+    "\n",
+    "    curobj.execute(con_validate_trigger)\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 22,
+   "id": "5dafecb2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def log_table_trigger():\n",
+    "    log_tbl_insert_trigger=\"create trigger IF NOT EXISTS insert_trigger\\\n",
+    "                        after insert \\\n",
+    "                        on contact\\\n",
+    "                        begin\\\n",
+    "                            insert into logtable\\\n",
+    "                            values(new.fname,new.lname,new.contact,datetime('now','localtime'),'INSERT');\\\n",
+    "                        end;\"\n",
+    "\n",
+    "    curobj.execute(log_tbl_insert_trigger)\n",
+    "    log_table_delete_trigger=\"create trigger IF NOT EXISTS delete_trigger\\\n",
+    "                                after delete \\\n",
+    "                                on contact\\\n",
+    "                                begin\\\n",
+    "                                insert into logtable\\\n",
+    "                                values(old.fname,old.lname,old.contact,datetime('now','localtime'),'DELETE');\\\n",
```

Comment:
whats .new, .old use for?

Reply:
The old is represented or used as already existing record and new is represented as the new record inserted after the old record . The new inserted record will overwrite the old record .